test/e2e/util.go

```diff
@@ -1157,22 +1172,27 @@ func waitForNodeToBe(c *client.Client, name string, wantReady bool, timeout time
 			continue
 		}
 
-		// Check the node readiness condition (logging all).
-		for i, cond := range node.Status.Conditions {
-			Logf("Node %s condition %d/%d: type: %v, status: %v",
-				name, i+1, len(node.Status.Conditions), cond.Type, cond.Status)
-			// Ensure that the condition type is readiness and the status
-			// matches as desired.
-			if cond.Type == api.NodeReady && (cond.Status == api.ConditionTrue) == wantReady {
-				Logf("Successfully found node %s readiness to be %t", name, wantReady)
-				return true
-			}
+		if isNodeReadySetAsExpected(node, wantReady) {
+			return true
 		}
 	}
 	Logf("Node %s didn't reach desired readiness (%t) within %v", name, wantReady, timeout)
 	return false
 }
 
+// Filters nodes in NodeList in place, removing nodes that do not
+// satisfy the given condition
+func filterNodes(nodeList *api.NodeList, fn func(node api.Node) bool) {
```

Comment:
Can you add a TODO: consider merging with pkg/client/cache.NodeLister

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/cache/listers.go#L111